File: QuotesforMindandSoul/app/src/main/java/ro/danserboi/quotesformindandsoul/QuoteDao.java
package ro.danserboi.quotesformindandsoul;

import androidx.room.Dao;
// import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface QuoteDao {

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insertQuote(Quote quote);

    // @Insert()
    // void insertQuoteList(List<Quote> quoteList);

    @Query("SELECT * FROM quotes_table WHERE words LIKE :words")
    List<Quote> getQuote(String words);

    @Query("SELECT * FROM quotes_table WHERE category LIKE :category")
    List<Quote> getQuotesByCategory(String category);

    @Query("SELECT * FROM quotes_table WHERE isBookmarked LIKE :isBookmarked")
    List<Quote> getQuotesByBookmark(Boolean isBookmarked);

    @Query("SELECT * FROM quotes_table")
    List<Quote> getAllQuotes();

    @Update
    void updateQuote(Quote quote);

    // @Delete
    // int deleteQuote(Quote quote);
}

File: QuotesforMindandSoul/app/src/main/java/ro/danserboi/quotesformindandsoul/QuoteDao.kt
package ro.danserboi.quotesformindandsoul

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy.Companion.IGNORE
import androidx.room.Query
import androidx.room.Update

@Dao
interface QuoteDao {
    @Insert(onConflict = IGNORE)
    fun insertQuote(quote: Quote)

    @Query("SELECT * FROM quotes_table WHERE words LIKE :words")
    fun getQuote(words: String): List<Quote>

    @Query("SELECT * FROM quotes_table WHERE category LIKE :category")
    fun getQuotesByCategory(category: String): List<Quote>

    @Query("SELECT * FROM quotes_table WHERE isBookmarked LIKE :isBookmarked")
    fun getQuotesByBookmark(isBookmarked: Boolean): List<Quote>

    @get:Query("SELECT * FROM quotes_table")
    val allQuotes: List<Quote>

    @Update
    fun updateQuote(quote: Quote)
}